src/conditions/pullRequestAuthorRole.ts

```diff
@@ -0,0 +1,20 @@
+import { ConditionConfig } from './../config'
+import { PullRequestInfo } from '../models'
+import { ConditionResult } from '../condition'
+import { getAssociationPriority } from '../association'
+
+export default function hasPullRequestAuthority (
+  config: ConditionConfig,
+  pullRequestInfo: PullRequestInfo
+): ConditionResult {
+  const associationString = ['NONE', 'FIRST_TIME', 'FIRST_TIME_CONTRIBUTOR', 'CONTRIBUTOR', 'COLLABORATOR', 'MEMBER', 'OWNER']
+
+  return associationString.indexOf(config.pullRequestAuthorRole) > getAssociationPriority(pullRequestInfo.authorAssociation)
```

Comment:
```suggestion
  return getAssociationPriority(config.pullRequestAuthorRole) > getAssociationPriority(pullRequestInfo.authorAssociation)
```

It isn't possible to reuse `getAssociationPriority` here?

Reply:
```suggestion
  return associationString.indexOf(config.pullRequestAuthorRole) >= getAssociationPriority(pullRequestInfo.authorAssociation)
```
It should be greater than or equal I think?

Reply:
To use `getAssociationPriority` I'd have to translate `pullRequestAuthorRole` from `string` to `CommentAuthorAssociation`. Is that what you're suggesting? Should I do it inside `config.ts`?

If they're equal, it should be a success, so it's `>`. The first test checks this.

Reply:
> If they're equal, it should be a success, so it's >. The first test checks this.
Ah sorry, my bad. That is correct.

> Should I do it inside config.ts?

Yes, the config type doesn't match what we expect the type to be. I made some suggestions in the new review.